src/modules/noit_lua_noit_binding.c

```diff
@@ -138,89 +139,10 @@ lua_noit_checks_unsubscribe(lua_State *L) {
   return lua_noit_checks_adjustsubscribe(L, -1);
 }
 
-static int
-noit_metric_value_index_func(lua_State *L) {
-  int n;
-  const char *k;
-  metric_value_t **udata, *metric_id;
-  n = lua_gettop(L); /* number of arguments */
-  assert(n == 2);
-  if(!luaL_checkudata(L, 1, "metric_value_t")) {
-    luaL_error(L, "metatable error, arg1 not a metric_value_t!");
-    return 1;
-  }
-  udata = lua_touserdata(L, 1);
-  metric_id = *udata;
-  if(!lua_isstring(L, 2)) {
-    luaL_error(L, "metatable error, arg2 not a string!");
-    return 1;
-  }
-
-  k = lua_tostring(L, 2);
-  switch (*k) {
-    case 'w':
-      if(!strcmp(k, "whence")) {
-        lua_pushinteger(L, metric_id->whence_ms / 1000);
-      } else {
-        break;
-      }
-      return 1;
-    case 't':
-      if(!strcmp(k, "type")) {
-        lua_pushinteger(L, metric_id->type);
-      } else {
-        break;
-      }
-      return 1;
-    case 'v':
-      if(!strcmp(k, "value")) {
-        switch (metric_id->type) {
-          case METRIC_INT32:
-            lua_pushinteger(L, metric_id->value.v_int32);
-            break;
-          case METRIC_INT64:
-            lua_pushinteger(L, metric_id->value.v_int64);
-            break;
-          case METRIC_UINT32:
-            lua_pushinteger(L, metric_id->value.v_int32);
-            break;
-          case METRIC_UINT64:
-            lua_pushinteger(L, metric_id->value.v_int64);
```

Comment:
that code is deleted! I've replaced it with ffi_libnoit.lua:

I'm using it this way:
`
self.numeric = ffi.new("noit_numeric_rollup_accu", {})

...

if self.numeric.accumulated.type == string.byte("i") then
    value =  self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("l") then
    value = self.numeric.accumulated.value.v_int64
  end

  if self.numeric.accumulated.type == string.byte("I") then
    value = self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("L") then
    value = self.numeric.accumulated.value.v_int32
  end

  if self.numeric.accumulated.type == string.byte("n") then
    value = self.numeric.accumulated.value.v_double
  end

  if self.numeric.accumulated.type == string.byte("s") then
    value = self.numeric.accumulated.value.v_string
  end
  if value ~= false then
    return tonumber(value)
  end
  return false
`